### PR TITLE
Use datacenter.iers.org as primary source for IERS data

### PR DESCRIPTION
**Pull request by tepickering**
### Description

This pull request configures the default URLs for downloading IERS data to point to datacenter.iers.org. While that site has had its own issues (see #12998 and discussion in #9227), it is the canonical source for data provided by the IERS.  ~Additionally, it was found that the `IERS_B` file at the original `IERS_B_URL` was almost a month out of date compared to the one provided by datacenter.iers.org.~

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/utils/iers/iers.py
Comment (pllim) [inline astropy/utils/iers/iers.py:49]: Is this number `_14_` going to change every time they update it?

**Comment by pllim** [comment]
Probably should backport all the way to LTS. And probably also need a change log, but let's wait for review. Thanks!

**Comment by pllim** [comment]
Also, if this becomes a blocker for v5.1 release, @saimn needs to know.

File: astropy/utils/iers/iers.py
Comment (pllim) [inline astropy/utils/iers/iers.py:48]: I think `IERS_B_FILE` points to a local file in https://github.com/astropy/astropy/tree/main/astropy/utils/iers/data . Do you really have to change this one?

File: astropy/utils/iers/iers.py
Comment (tepickering) [reply astropy/utils/iers/iers.py:49]: it's in the `c04_14` section under `long-term` so i don't think so. it's at least been stable since 2017, fwiw.

File: astropy/utils/iers/iers.py
Comment (tepickering) [reply astropy/utils/iers/iers.py:48]: yeah, the different filenames between the mirrors is annoying. is there a reason `IERS_B` points to packaged data (which may be well out of date) while `IERS_A` does not and is always downloaded?

Comment (tepickering) [reply astropy/utils/iers/iers.py:48]: anyway, looks like a deeper can of worms there so i'll revert it and leave `IERS_B` alone for now.

File: astropy/utils/iers/iers.py
Comment (pllim) [inline astropy/utils/iers/iers.py:50]: Is this new mirror used anywhere?

Comment (tepickering) [reply astropy/utils/iers/iers.py:50]: no, but neither is `IERS_B_URL`. only the packaged data is ever used for `IERS_B`.

**Comment by pllim** [comment]
Maaaybe this PR should stick with just swapping IERS A URLs and IERS B should be a separate PR... https://github.com/astropy/astropy/search?q=IERS_B_URL

**Comment by mhvk** [comment]
The `IERS_B` file is included since it is "final" data, so useful for anything historical (i.e., most data people would be analyzing); IERS_A is downloaded as needed since that includes predictions for the near future.  The IERS B URLs are mostly just so that people know how they can update that if needed; I think they are mentioned in the docs.

For this PR, I would indeed suggest to focus on IERS A, since that is the one that gives the problems!

**Comment by tepickering** [comment]
yeah, i agree with these points.

**Comment by tepickering** [comment]
i should note that in addition to containing more recent data, the IERS version of `IERS_B_FILE` has some (very) slight differences in the historical data. part of a separate PR for updating `IERS_B_URL` should be confirming the provenance of the data and who gets what from there. the file cites obspm.fr as the source, but the data from the IERS datacenter is clearly newer...

**Comment by pllim** [comment]
@mhvk , do we need a change log for this?

@tepickering , eventually, this will need a squash before merge. FYI.

**Comment by mhvk** [comment]
I don't think a changelog entry is necessary - the point of having those variables is exactly what we are doing here, that we can adjust them as needed!

**Review by mhvk — APPROVED** [review]
Thanks!!

**Comment by pllim** [comment]
Thanks, all!

@meeseeksdev backport to v5.0.x

**Comment by pllim** [comment]
@meeseeksdev backport to v5.1.x

**Pull request merged**
